diff -up texlive-2007/texk/web2c/tex.ch.than texlive-2007/texk/web2c/tex.ch
--- texlive-2007/texk/web2c/tex.ch.than	2014-04-04 15:47:57.511874312 +0200
+++ texlive-2007/texk/web2c/tex.ch	2014-04-04 15:49:29.855429545 +0200
@@ -2210,12 +2210,20 @@ var k:0..buf_size; {index into |buffer|}
 @y
 var k:0..buf_size; {index into |buffer|}
 @!saved_cur_name:str_number; {to catch empty terminal input}
+@!saved_cur_ext:str_number; {to catch empty terminal input}
+@!saved_cur_area:str_number; {to catch empty terminal input}
 @z
 
 @x [29.530] l.10252 - prompt_file_name: No default extension is TeX input file.
 if e=".tex" then show_context;
 @y
 if (e=".tex") or (e="") then show_context;
+print_ln; print("(Press Enter to retry, or Control-D to exit");
+if (e<>"") then
+  begin
+    print("; default file extension is `"); print(e); print("'");
+  end;
+print(")"); print_ln;
 @z
 
 @x [29.530] l.10258 - prompt_file_name: prevent empty filenames.
@@ -2223,9 +2231,17 @@ clear_terminal; prompt_input(": "); @<Sc
 if cur_ext="" then cur_ext:=e;
 @y
 saved_cur_name:=cur_name;
+saved_cur_ext:=cur_ext;
+saved_cur_area:=cur_area;
 clear_terminal; prompt_input(": "); @<Scan file name in the buffer@>;
-if cur_ext="" then cur_ext:=e;
-if length(cur_name)=0 then cur_name:=saved_cur_name;
+if (length(cur_name)=0) and (cur_ext="") and (cur_area="") then
+  begin
+    cur_name:=saved_cur_name;
+    cur_ext:=saved_cur_ext;
+    cur_area:=saved_cur_area;
+  end
+else
+  if cur_ext="" then cur_ext:=e;
 @z
 
 @x [29.532] l.10263 - avoid conflict, `logname' in <unistd.h> on some systems.
