diff -up texmf-dist/scripts/tetex/updmap.pl.me texmf-dist/scripts/tetex/updmap.pl
--- texmf-dist/scripts/tetex/updmap.pl.me	2018-07-11 17:04:20.267484075 +0200
+++ texmf-dist/scripts/tetex/updmap.pl	2018-07-11 17:04:05.611892683 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 # mupdmap: utility to maintain map files for outline fonts.
 # $Id: updmap.pl 27264 2012-07-31 05:10:29Z preining $
 # 
@@ -2071,8 +2071,8 @@ sub reset_root_home {
       my (undef,undef,undef,undef,undef,undef,undef,$roothome) = getpwuid(0);
       if (defined($roothome)) {
         if ($envhome ne $roothome) {
-          warning("$prg: resetting \$HOME value (was $envhome) to root's "
-            . "actual home ($roothome).\n");
+          # warning("$prg: resetting \$HOME value (was $envhome) to root's "
+          #  . "actual home ($roothome).\n");
           $ENV{'HOME'} = $roothome;
         } else {
           # envhome and roothome do agree, nothing to do, that is the good case
