diff -up texlive-2012/source/texk/kpathsea/c-auto.in.than texlive-2012/source/texk/kpathsea/c-auto.in
diff -up texlive-2012/source/texk/kpathsea/configure.than texlive-2012/source/texk/kpathsea/configure
--- texlive-2012/source/texk/kpathsea/configure.than	2013-11-13 16:00:54.707154960 +0100
+++ texlive-2012/source/texk/kpathsea/configure	2013-11-13 16:11:07.692156501 +0100
@@ -13268,7 +13268,12 @@ $as_echo "$ac_cv_sizeof_long" >&6; }
 
 
 cat >>confdefs.h <<_ACEOF
-#define SIZEOF_LONG $ac_cv_sizeof_long
+#include <bits/wordsize.h>
+#if __WORDSIZE == 32
+#define SIZEOF_LONG 4
+#else
+#define SIZEOF_LONG 8
+#endif
 _ACEOF
 
 
